### PR TITLE
reformatted tracing-api.md and added TOC.

### PR DESCRIPTION
**Pull request by rghetia**
unified heading format. Some header had just name while others had return values.
- Changed it to just name.

header levels were different (some with ### and others with ####) under same section.
- Fixed this.

Added TOC.

### CODE REVIEW

File: specification/tracing-api.md [hunk 1]
```diff
@@ -1,3 +1,55 @@
+# Table of Content
+<details>
```
Comment (SergeyKanzhelev) [inline specification/tracing-api.md]: wow! this is cool

**Comment by SergeyKanzhelev** [comment]
@rghetia please use the same email for commits as registered for CNCF CLA

**Comment by rghetia** [comment]
I signed it

**Comment by SergeyKanzhelev** [comment]
> I signed it

this happens when you push commit with the e-mail that is different from the one that is default in your GitHub account and that you used to sign CLA. run `got log -1` to check what e-mail was used in commit

**Comment by rghetia** [comment]
> > I signed it
> 
> this happens when you push commit with the e-mail that is different from the one that is default in your GitHub account and that you used to sign CLA. run `got log -1` to check what e-mail was used in commit

it is different email. Let me close this and open a new one.

**Pull request closed without merging**